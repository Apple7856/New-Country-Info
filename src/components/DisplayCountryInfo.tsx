import { Button, Container, Paper, SvgIcon, Typography } from '@material-ui/core';
import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

const DisplayCountryInfo = () => {
    const location = useLocation();
    const [countryData, setCountryData] = useState<any>("");
    const [flag, setFlag] = useState<any>("");
    const [capitalData, setCapitalData] = useState<any>("");
    const [showWeather, setShowWeather] = useState(false);
    console.log(location.state);
    useEffect(() => {
        axios({
            method: "get",
            url: `https://restcountries.com/v2/name/${location.state}`,
            responseType: "stream"
        })
            .then((res) => {
                // console.log(res.data[1].flags);
                setFlag(res.data[1].flags);
                setCountryData(res.data[1]);
            })
            .catch((error) => {
                console.log(error);
            })
    }, [])
    const getCapitalWeather = async () => {
        try {
            const responce = await axios.get(`http://api.weatherstack.com/current?access_key=799d2cb12c98d1978925b8fa886e1e0a&query=${countryData.capital}`)
            setCapitalData(responce.data.current);
            setShowWeather(true)
        } catch (error) {
            console.log(error);
        }
    }
    return <Container>
        <Paper>
            <Typography variant='h2' data-testid="heading">Country Info</Typography>
            <Typography variant='body1'>Capital: {countryData.capital}</Typography>
            <Typography variant='body1'>Population: {countryData.population}</Typography>
            <Typography variant='body1'>latlng: {countryData.latlng}</Typography>
            <Typography variant='body1'>Flag: <img src={flag.svg} width="100px" /></Typography>
            <Button variant='contained' color="primary" onClick={() => getCapitalWeather()}>Capital Weather</Button>
        </Paper>
        {
            showWeather ?
                <>
                    <Typography variant='h2'>Capital Weather</Typography>

                    <Typography variant='body1'>temperature: {capitalData.temperature}</Typography>
                    <Typography variant='body1'>weather_icons: <img src={capitalData.weather_icons} width="100px" /></Typography>
                    <Typography variant='body1'>Wind_speed: {capitalData.wind_speed}</Typography>
                    <Typography variant='body1'>Precip: {capitalData.precip}</Typography>
                </>
                : ""
        }
    </Container>;
};

export default DisplayCountryInfo;
