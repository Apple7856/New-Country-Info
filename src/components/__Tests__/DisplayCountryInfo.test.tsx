import DisplayCountryInfo from '../DisplayCountryInfo';
import { cleanup, render } from '@testing-library/react';
import renderer, { act } from 'react-test-renderer';
import axios from 'axios';

beforeEach(() => {
    document.body.innerHTML = ""
})

afterEach(() => {
    cleanup()
});

const mockedLocation = jest.fn();

jest.mock('react-router-dom', () => {
    const originalRouterDom = jest.requireActual('react-router-dom');
    return {
        __esModule: true,
        ...originalRouterDom,
        useLocation: () => mockedLocation
    };
});

describe("SearchCountry Component", () => {
    test("Heading test", () => {
        const { getByTestId } = render(<DisplayCountryInfo />);
        const heading = getByTestId('heading');
        expect(heading).toBe("Country Info");
    })

    test("Snapshot test", () => {
        const component = renderer.create(<DisplayCountryInfo />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("should return users list", async () => {
        await act(async () => {
            const responce = await axios.get("https://restcountries.com/v2/name/india");
            expect(responce.data).toBeDefined();
        })
    });
})
