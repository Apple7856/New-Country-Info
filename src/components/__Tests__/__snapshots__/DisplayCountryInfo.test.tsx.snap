// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCountry Component Snapshot test 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2"
      data-testid="heading"
    >
      Country Info
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Capital: 
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Population: 
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      latlng: 
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Flag: 
      <img
        width="100px"
      />
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Capital Weather
      </span>
    </button>
  </div>
  
</div>
`;
